runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/RequestCompressionTraitInterceptor.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.interceptors
+
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.client.ProtocolRequestInterceptorContext
+import aws.smithy.kotlin.runtime.http.interceptors.requestcompression.CompressionAlgorithm
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+import aws.smithy.kotlin.runtime.http.request.HttpRequestBuilder
+import aws.smithy.kotlin.runtime.http.request.header
+import aws.smithy.kotlin.runtime.http.request.toBuilder
+
+private val VALID_COMPRESSION_RANGE = 0..10485760
+
+/**
+ * HTTP interceptor that compresses operation request payloads when eligible
+ */
+@InternalApi
+public class RequestCompressionTraitInterceptor(
+    private val compressionThreshold: Int,
+    private val requestedCompressionAlgorithms: List<String>,
+    private val supportedCompressionAlgorithms: List<CompressionAlgorithm>,
```

Comment:
question: I think these are swapped [based on your usage in **aws-sdk-kotlin**](https://github.com/awslabs/aws-sdk-kotlin/pull/1120/files#diff-dea08dda1e260a9c2df5faed3bfe24787a318c0b121291ee5c2129791f223df8R70-R83)? Does the client config supply the "requested"  algorithms and the modeled trait provides the "supported" algorithms?